pkg/controllers/state/cluster.go

```diff
@@ -464,6 +476,19 @@ func (c *Cluster) updatePod(ctx context.Context, pod *v1.Pod) error {
 		err = c.updateNodeUsageFromPod(ctx, pod)
 	}
 	c.updatePodAntiAffinities(pod)
+	if pod.DeletionTimestamp != nil {
+		// We could do a more complex cmp/swap operation, but since deletion/creation timestamps have a resolution of
```

Comment:
nit: Sorry, maybe I'm being confusing. I just meant that this comment should note that there is a potential race here. I personally don't think you need to go into full detail into how we could improve.

Reply:
Sorry, I think it's fine as is.